### DISCUSSION TITLE
Is it possible to define dependencies conditional on environment variables?

### DISCUSSION THREAD

**Discussion by paulbeckmann** [go-task/task#800] [Q&A]
Hi there, 
I was wondering whether there is any way to define dependencies depending on e.g. the value of an environment variable. 

I have played around with preconditions but cannot get it to work. 

In essence, I am looking for a way to tell a task that it should run either "deps: [option_a]" or "deps: [option_b]" depending on the value of an environment variable.  

I appreciate any support on this!

**Comment by ghostsquad** [comment]
I would create 2 dependency tasks, depend on both.

The individual tasks should use "status" (not precondition).

Status should always "succeed" on the wrong architecture in order to prevent that from running.

Each architecture can have subsequent status steps to influence up-to-date-ness.
